ports/gstplugin/servosrc.rs

```diff
@@ -0,0 +1,566 @@
+            debug_assert_eq!(
+                gfx.gl.check_framebuffer_status(gl::FRAMEBUFFER),
+                gl::FRAMEBUFFER_COMPLETE
+            );
+        })
+    }
+
+    fn gl(&self) -> Rc<dyn gleam::gl::Gl> {
+        self.gl.clone()
+    }
+
+    fn get_coordinates(&self) -> EmbedderCoordinates {
+        GFX.with(|gfx| {
+            debug!("EMBEDDER get_coordinates");
+            let mut gfx = gfx.borrow_mut();
+            let gfx = &mut *gfx;
+            let size = gfx
+                .device
+                .context_surface_info(&gfx.context)
+                .unwrap()
+                .unwrap()
+                .size;
+            let size = Size2D::from_untyped(size);
+            let origin = Point2D::origin();
+            debug_assert_eq!(gfx.gl.get_error(), gl::NO_ERROR);
+            EmbedderCoordinates {
+                hidpi_factor: Scale::new(1.0),
+                screen: size,
+                screen_avail: size,
+                window: (size, origin),
+                framebuffer: size,
+                viewport: Rect::new(origin, size),
+            }
+        })
+    }
+
+    fn set_animation_state(&self, _: AnimationState) {}
+
+    fn get_gl_context(&self) -> servo_media::player::context::GlContext {
+        servo_media::player::context::GlContext::Unknown
+    }
+
+    fn get_native_display(&self) -> servo_media::player::context::NativeDisplay {
+        servo_media::player::context::NativeDisplay::Unknown
+    }
+
+    fn get_gl_api(&self) -> servo_media::player::context::GlApi {
+        servo_media::player::context::GlApi::OpenGL3
+    }
+}
+
+impl ObjectSubclass for ServoSrc {
+    const NAME: &'static str = "ServoSrc";
+    // gstreamer-gl doesn't have support for GLBaseSrc yet
+    // https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/issues/219
+    type ParentType = BaseSrc;
+    type Instance = ElementInstanceStruct<Self>;
+    type Class = ClassStruct<Self>;
+
+    fn new() -> Self {
+        let (sender, receiver) = crossbeam_channel::bounded(1);
+        thread::spawn(move || ServoThread::new(receiver).run());
+        let (acks, ackr) = crossbeam_channel::bounded(1);
+        let _ = sender.send(ServoSrcMsg::GetSwapChain(acks));
+        let swap_chain = ackr.recv().expect("Failed to get swap chain");
+        let info = Mutex::new(None);
+        Self {
+            sender,
+            swap_chain,
+            info,
+        }
+    }
+
+    fn class_init(klass: &mut ClassStruct<Self>) {
+        klass.set_metadata(
+            "Servo as a gstreamer src",
+            "Filter/Effect/Converter/Video",
+            "The Servo web browser",
+            env!("CARGO_PKG_AUTHORS"),
+        );
+
+        let src_caps = Caps::new_simple(
+            "video/x-raw",
+            &[
+                ("format", &VideoFormat::Bgrx.to_string()),
+                ("width", &IntRange::<i32>::new(512, 1024)),
+                ("height", &IntRange::<i32>::new(512, 1024)),
```

Comment:
Why limited to these resolutions instead of `(1, i32::MAX)`?

Reply:
Good poimt, that was just for testing. Fixed.